socorro/stage_submitter/submitter.py

```diff
@@ -0,0 +1,455 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+"""
+This defines the stage submitter application. It's designed to run as a standalone
+service.
+
+It consumes crash ids from a queue, determines what to do with them, pulls the crash
+data from storage, assembles a payload, and submits them to a destination.
+
+It pulls all its configuration from socorro.settings reusing processor configuration
+where convenient.
+
+To run::
+
+    $ /app/bin/run_submitter.sh
+
+"""
+
+from contextlib import suppress
+import dataclasses
+from email.header import Header
+import gzip
+import json
+import io
+import logging
+import pathlib
+import random
+import sys
+import traceback
+from typing import Callable
+
+from fillmore.libsentry import set_up_sentry
+from fillmore.scrubber import Scrubber, SCRUB_RULES_DEFAULT
+import markus
+import requests
+import sentry_sdk
+
+from socorro import settings
+from socorro.external.crashstorage_base import CrashIDNotFound
+from socorro.lib.libdockerflow import get_release_name, get_version_info
+from socorro.lib.liblogging import set_up_logging
+from socorro.libclass import build_instance_from_settings
+
+
+# Metrics client to use
+METRICS = markus.get_metrics("socorro.submitter")
+
+# Default user agent to use when submitting to a destination url
+DEFAULT_USER_AGENT = "stage-submitter/2.0"
+
+# Boundary to use. This is the result of uuid.uuid4().hex. We just need a unique string
+# to denote the boundary between parts in the payload.
+BOUNDARY = "01659896d5dc42cabd7f3d8a3dcdd3bb"
+
+
+@dataclasses.dataclass
+class Crash:
+    crash_id: str
+    finished_func: Callable[[], None]
+
+
+@dataclasses.dataclass
+class Destination:
+    url: str
+    sample: int
+
+
+def get_destinations(value_list):
+    """Parse Destination instances from list of strings
+
+    :arg value_list: list of `|` delimited strings
+
+    :returns: list of Destination instances
+
+    """
+    destinations = []
+    for item in value_list:
+        url, sample = item.split("|")
+        destinations.append(Destination(url=url, sample=int(sample)))
+    return destinations
+
+
+COLLECTOR_KEYS_TO_REMOVE = [
+    "metadata",
+    "submitted_timestamp",
+    "version",
+]
+
+
+def remove_collector_keys(raw_crash):
```

Comment:
Large portions of this file were copied verbatim from socorro-submitter repository where they've been running for years.